pkg/config/config.go

```diff
@@ -48,6 +67,13 @@ type ComponentSettings struct {
 
 // LocalConfig holds all the config relavent to a specific Component.
 type LocalConfig struct {
+	typeMeta          metav1.TypeMeta   `yaml:",inline"`
```

Comment:
Now `LocalConfig` and `proxyLocalConfig` have complety different structures.

`proxyLocalConfig` has `metav1.TypeMeta` embeded, and `LocalConfig` has it under the `typMeta`

```
LocalConfig {
  typeMeta: {
      kind: "LocalConfig",
      apiversion: "odo.openshift.io/v1alpha1"
  },
  componentSettings {
      Name: "mycomponent",
       .....
  }
}
```
vs 
```
proxyLocalConfig {
  kind: "LocalConfig",
  apiversion: "odo.openshift.io/v1alpha1"
  componentSettings {
      Name: "mycomponent",
       .....
  }
}
```

Is that what you wanted?

Reply:
to hide the `typemeta` that's the only way @kadel

Reply:
and that's what I said in one of the comments that both `localconfig` and `proxylocalconfig` need to have same structure which will not be possible if we hide the `typemeta`

Reply:
But we could still remove it, no?? 
What is the reason you want to keep it there?

Reply:
https://github.com/redhat-developer/odo/pull/1453#discussion_r265453192

Took the 2nd approach

Reply:
I want the localconfig to be the primary source of truth, proxyconfig is just a transient struct

Reply:
And even the proxyconfig gets the type meta from localconfig, so in my opinion we should keep it. But I will keep the final decision to you